apis/rust/node/src/node/mod.rs

```diff
@@ -66,6 +95,63 @@ impl DoraNode {
         Self::init(node_config)
     }
 
+    /// Initiate a node from a dataflow id and a node id.
+    ///
+    /// ```no_run
+    /// use dora_node_api::DoraNode;
+    /// use dora_node_api::dora_core::config::NodeId;
+    ///
+    /// let (mut node, mut events) = DoraNode::init_from_node_id(None, NodeId::from("plot".to_string()), None).expect("Could not init node plot");
+    /// ```
+    ///
+    pub fn init_from_node_id(
+        dataflow_uuid: Option<String>,
+        node_id: NodeId,
+        tracing: Option<bool>,
+    ) -> eyre::Result<(Self, EventStream)> {
+        // Make sure that the node is initialized outside of dora start.
+        if let Ok(node_config_string) = std::env::var("DORA_NODE_CONFIG") {
+            let node_config: NodeConfig = serde_yaml::from_str(&node_config_string)
+                .context("failed to deserialize operator config")?;
+            assert!(
+                node_config.node_id == node_id,
+                "Node id within the yaml description and the node_id does not match. Please either run this node in either detached mode or change or remove `node_id` specification in the code."
+            );
+            return Self::init(node_config);
+        }
+
+        let mut session = TcpLayer::new()
+            .connect(control_socket_addr())
+            .context("could not connect to the coordinator")?;
```

Comment:
This only works when the coordinator runs on the same machine. For distributed deployments, every node would need to know the dataflow IP and port using this approach.

If we communicate through the daemon instead, only the daemon needs to know how to reach the coordinator.

Reply:
Got it